src/rgw: Adding null version id support

TRYTOBE8TME opened the pull request:
<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

TRYTOBE8TME commented:
Still working on adding test cases. Will update soon.

smanjara commented:
the commits could be squashed into a single one or into multiple ones with meaningful commit messages.

also, as discussed, we may not have to introduce a new variable in the `rgw_bi_log_entry` if we add a flag. we already have `bilog_flags` that we could make use of. this avoids the cls encode/decode changes.
I have an experimental branch here https://github.com/ceph/ceph/commit/de71f4f609a7e70c59022f95adc1b4e25bf66fae which has this alternate version that extends bilog_flags, but I haven't been able to test it. Would you mind taking a look? Mostly everything remains the same. we just don't encode anything.

TRYTOBE8TME commented:
> the commits could be squashed into a single one or into multiple ones with meaningful commit messages.
> 
> also, as discussed, we may not have to introduce a new variable in the `rgw_bi_log_entry` if we add a flag. we already have `bilog_flags` that we could make use of. this avoids the cls encode/decode changes. I have an experimental branch here [de71f4f](https://github.com/ceph/ceph/commit/de71f4f609a7e70c59022f95adc1b4e25bf66fae) which has this alternative version that extends bilog_flags, but I haven't been able to test it. Would you mind taking a look? Mostly everything remains the same. we just don't encode anything.

Sure, I'll squash the commits into one and I'll test your branch as well. Thanks for this!

cbodley commented:
what bug is this fixing? i don't see any commit messages or tracker/bz links

smanjara commented:
> what bug is this fixing? i don't see any commit messages or tracker/bz links

this is supposed to be [bz#1967576](https://bugzilla.redhat.com/show_bug.cgi?id=1967576) and [tracker#55179](https://tracker.ceph.com/issues/55179)

smanjara commented:
@TRYTOBE8TME I have rebased and updated my branch [wip-shilpa-null-version](https://github.com/smanjara/ceph/tree/wip-shilpa-null-version) with a working version of this. I tested some basic cases and they work for me. Can you take a look? You could pull my changes into this one or open a new PR. upto you.

TRYTOBE8TME commented:
> @TRYTOBE8TME I have rebased and updated my branch [wip-shilpa-null-version](https://github.com/smanjara/ceph/tree/wip-shilpa-null-version) with a working version of this. I tested some basic cases and they work for me. Can you take a look? You could pull my changes into this one or open a new PR. upto you.

On it, will update soon!
